sotodlib/tod_ops/interp_drop.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+from sotodlib import core
+from so3g.proj import Ranges
+
+def interp_drop_array(arr, drop_idxes):
+    """
+    Function to interpolate numpy.array.
+    Insert the mean of left and right samples at the dropped indexes
+    
+    Args:
+        arr: array contains data drops.
+        drop_idxes: Indexes of dropped samples.
+        
+    Return:
+        new_arr: array that drops are filled.
+    """
+    data_dim = len(np.shape(arr))
+    new_arr = np.copy(arr)
+    
+    for  i in range(len(drop_idxes)):
+        drop_idx = drop_idxes[i] + i
+        if data_dim == 1:
+            # Such as azimuth or elevation
+            inserted_data = (new_arr[drop_idx-1] + new_arr[drop_idx])/2.
+            new_arr = np.insert(new_arr, drop_idx, inserted_data)
+            
+        elif data_dim == 2:
+            inserted_data = (new_arr[:, drop_idx-1] + new_arr[:, drop_idx])/2.
+            new_arr = np.insert(new_arr, drop_idx, inserted_data, axis=1)
```

Comment:
In case you're not familiar with it -- you can do this for arbitrary ndim using the "..." super index, e.g.:
```
  inserted_data = (new_arr[..., drop_idx-1] + new_arr[..., drop_idx]) / 2
```
That should work ndim >= 1, actually.

Reply:
I did not know that. That looks great!